ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionValidator.java

```diff
@@ -40,24 +42,46 @@
   private final Optional<BigInteger> chainId;
 
   private Optional<TransactionFilter> transactionFilter = Optional.empty();
+  private final Optional<EIP1559> maybeEip1559;
 
   public MainnetTransactionValidator(
       final GasCalculator gasCalculator,
       final boolean checkSignatureMalleability,
       final Optional<BigInteger> chainId) {
+    this(gasCalculator, checkSignatureMalleability, chainId, Optional.empty());
+  }
+
+  public MainnetTransactionValidator(
+      final GasCalculator gasCalculator,
+      final boolean checkSignatureMalleability,
+      final Optional<BigInteger> chainId,
+      final Optional<EIP1559> maybeEip1559) {
     this.gasCalculator = gasCalculator;
     this.disallowSignatureMalleability = checkSignatureMalleability;
     this.chainId = chainId;
+    this.maybeEip1559 = maybeEip1559;
   }
 
   @Override
-  public ValidationResult<TransactionInvalidReason> validate(final Transaction transaction) {
+  public ValidationResult<TransactionInvalidReason> validate(
+      final Transaction transaction, final long blockNumber) {
```

Comment:
Why do we need the `blockNumber`?  We get a separate TransactionValidator per fork, so either EIP 1559 will be configured because it applies in that fork or it won't be present because it wasn't configured. It's all kept in the ProtocolSchedule.  I think we should do without it.

Reply:
Oh yes you're right.

Reply:
Done.